### ISSUE TITLE
NotNullViolation when trying to import a learner account

### ISSUE THREAD

**Issue by intelliant01** [learningequality/kolibri#9139]
## Observed behavior

When trying to sync a learner account from a Kolibri 0.15.1 server with a Postgres backend, to the learner's app (Android or Windows) 0.15.1, it fails.
Last 2 things I see in the Android app are - 
`Remotely preparing data` followed by `2 of 7: Failed`

No facility imports are working when using this upgraded server as the source. They were working at least prior to upgrade to 0.15.x. I am not sure if they were even working after having upgraded to 0.15.0.

## Errors and logs

This is what's present in kolibri.log -
```
ERROR 2022-02-24 11:35:54,204 morango.sync.controller null value in column "transfer_stage_status" violates not-null constraint
DETAIL:  Failing row contains (0f15bf12-fa38-45e8-ac38-2e87de12c6af, 156205118be9ebea414630547bd83133:user-rw:d777ca55f496ac49861b210..., f, t, null, fe3e0240-e835-4e6e-ad8c-e6f77987aa1b, 2022-02-24 06:05:54.182339+00, {"super": {}, "sub": {}}, 0, {}, 2022-02-24 06:05:54.182318+00, 0, 0, initializing, null).

ERROR 2022-02-24 11:35:54,290 django.request Internal Server Error: /api/morango/v1/transfersessions/
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/kolibri/dist/morango/sync/operations.py", line 776, in handle
    transfer_session = transfer_sessions.get(**data)
  File "/usr/lib/python3/dist-packages/kolibri/dist/django/db/models/query.py", line 380, in get
    self.model._meta.object_name
morango.models.core.DoesNotExist: TransferSession matching query does not exist.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/kolibri/dist/django/db/backends/utils.py", line 64, in execute
    return self.cursor.execute(sql, params)
psycopg2.errors.NotNullViolation: null value in column "transfer_stage_status" violates not-null constraint
DETAIL:  Failing row contains (0f15bf12-fa38-45e8-ac38-2e87de12c6af, 156205118be9ebea414630547bd83133:user-rw:d777ca55f496ac49861b210..., f, t, null, fe3e0240-e835-4e6e-ad8c-e6f77987aa1b, 2022-02-24 06:05:54.182339+00, {"super": {}, "sub": {}}, 0, {}, 2022-02-24 06:05:54.182318+00, 0, 0, initializing, null).


The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/kolibri/dist/django/core/handlers/exception.py", line 41, in inner
    response = get_response(request)
  File "/usr/lib/python3/dist-packages/kolibri/dist/django/core/handlers/base.py", line 187, in _get_response
    response = self.process_exception_by_middleware(e, request)
  File "/usr/lib/python3/dist-packages/kolibri/dist/django/core/handlers/base.py", line 185, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/usr/lib/python3/dist-packages/kolibri/dist/django/views/decorators/csrf.py", line 58, in wrapped_view
    return view_func(*args, **kwargs)
  File "/usr/lib/python3/dist-packages/kolibri/dist/rest_framework/viewsets.py", line 116, in view
    return self.dispatch(request, *args, **kwargs)
  File "/usr/lib/python3/dist-packages/kolibri/dist/rest_framework/views.py", line 495, in dispatch
    response = self.handle_exception(exc)
  File "/usr/lib/python3/dist-packages/kolibri/dist/rest_framework/views.py", line 455, in handle_exception
    self.raise_uncaught_exception(exc)
  File "/usr/lib/python3/dist-packages/kolibri/dist/rest_framework/views.py", line 492, in dispatch
    response = handler(request, *args, **kwargs)
  File "/usr/lib/python3/dist-packages/kolibri/dist/morango/api/viewsets.py", line 392, in create
    raise context.error
  File "/usr/lib/python3/dist-packages/kolibri/dist/morango/sync/controller.py", line 236, in _invoke_middleware
    result = middleware(context)
  File "/usr/lib/python3/dist-packages/kolibri/dist/morango/registry.py", line 227, in __call__
    result = operation(context)
  File "/usr/lib/python3/dist-packages/kolibri/dist/morango/sync/operations.py", line 719, in __call__
    result = self.handle(context)
  File "/usr/lib/python3/dist-packages/kolibri/dist/morango/sync/operations.py", line 804, in handle
    transfer_session.save()
  File "/usr/lib/python3/dist-packages/kolibri/dist/django/db/models/base.py", line 808, in save
    force_update=force_update, update_fields=update_fields)
  File "/usr/lib/python3/dist-packages/kolibri/dist/django/db/models/base.py", line 838, in save_base
    updated = self._save_table(raw, cls, force_insert, force_update, using, update_fields)
  File "/usr/lib/python3/dist-packages/kolibri/dist/django/db/models/base.py", line 924, in _save_table
    result = self._do_insert(cls._base_manager, using, fields, update_pk, raw)
  File "/usr/lib/python3/dist-packages/kolibri/dist/django/db/models/base.py", line 963, in _do_insert
    using=using, raw=raw)
  File "/usr/lib/python3/dist-packages/kolibri/dist/django/db/models/manager.py", line 85, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
  File "/usr/lib/python3/dist-packages/kolibri/dist/django/db/models/query.py", line 1079, in _insert
    return query.get_compiler(using=using).execute_sql(return_id)
  File "/usr/lib/python3/dist-packages/kolibri/dist/django/db/models/sql/compiler.py", line 1112, in execute_sql
    cursor.execute(sql, params)
  File "/usr/lib/python3/dist-packages/kolibri/dist/django/db/backends/utils.py", line 64, in execute
    return self.cursor.execute(sql, params)
  File "/usr/lib/python3/dist-packages/kolibri/dist/django/db/utils.py", line 94, in __exit__
    six.reraise(dj_exc_type, dj_exc_value, traceback)
  File "/usr/lib/python3/dist-packages/kolibri/dist/django/utils/six.py", line 685, in reraise
    raise value.with_traceback(tb)
  File "/usr/lib/python3/dist-packages/kolibri/dist/django/db/backends/utils.py", line 64, in execute
    return self.cursor.execute(sql, params)
django.db.utils.IntegrityError: null value in column "transfer_stage_status" violates not-null constraint
DETAIL:  Failing row contains (0f15bf12-fa38-45e8-ac38-2e87de12c6af, 156205118be9ebea414630547bd83133:user-rw:d777ca55f496ac49861b210..., f, t, null, fe3e0240-e835-4e6e-ad8c-e6f77987aa1b, 2022-02-24 06:05:54.182339+00, {"super": {}, "sub": {}}, 0, {}, 2022-02-24 06:05:54.182318+00, 0, 0, initializing, null).

```

[Full kolibri.txt also herein](https://github.com/learningequality/kolibri/files/8130167/kolibri.txt)

## Expected behavior

The import should have worked without any errors.

## User-facing consequences

Unable to use the app.

## Steps to reproduce

Had upgraded this server from 0.14.x to 0.15.0 a few days ago and now to 0.15.1 .
Installed the "early access" Android app from play store and attempted to import.

## Context

kolibri-server 0.4.0
kolibri 0.15.1
Ubuntu 18.04.3
Postgres 12.3
Kolibri Android app or Windows app 0.15.1 
Python 3.6.9
Server  nginx/1.14.0 (Ubuntu)

**Comment by bjester** [comment]
@intelliant01 Thank you for submitting this issue. The Kolibri logs you've shared are from the server, correct?

**Comment by intelliant01** [comment]
@bjester yes. 

Is there a way to rollback this change to the postgres database or a query to fix this, because unfortunately I did not take a backup prior to the upgrade?

**Comment by bjester** [comment]
@intelliant01 From the logs, it seems that the database migrations didn't occur during the upgrade. Would you mind stopping Kolibri and running `kolibri manage migrate`? Please share the output, and of course start the server again afterwards.

**Comment by intelliant01** [comment]
~~My bad. The logs from the upgrade run are attached herein now which indicate that the migrations did run.~~

~~[kolibri-2022-02-24.txt](https://github.com/learningequality/kolibri/files/8165778/kolibri-2022-02-24.txt)~~

The output of `kolibri manage migrate` is -
```
$ kolibri manage migrate
WARNING  Option CHERRYPY_START in section [Server] is deprecated, please remove it from your options.ini file
INFO     Option DEBUG in section [Server] being overridden by environment variable KOLIBRI_DEBUG
INFO     Option DEBUG_LOG_DATABASE in section [Server] being overridden by environment variable KOLIBRI_DEBUG_LOG_DATABASE
INFO     Running Kolibri with the following settings: kolibri.deployment.default.settings.base
INFO     Importing 'tasks' module from django apps
INFO     Invoking command migrate
Operations to perform:
  Apply all migrations: admin, analytics, auth, bookmarks, content, contenttypes, device, discovery, exams, kolibriauth, lessons, logger, morango, notifications, oidc_provider, sessions
Running migrations:
  No migrations to apply.
```

An attempt to sync thereafter results in the same error as reported earlier.

**Comment by bjester** [comment]
@intelliant01 Prior to installing 0.15.0, did you install any 0.15.0 alpha/beta release? We should be able to correct this with some manual commands which I'm determining at the moment.

**Comment by bjester** [comment]
Alright, these commands should resolve your issue, but if not, please provide the output. I recommend stopping Kolibri first.
```
# rollback to migration prior to the one that isn't applied
kolibri manage migrate morango 0018
# reapply migrations
kolibri mange migrate
```

**Comment by intelliant01** [comment]
> @intelliant01 Prior to installing 0.15.0, did you install any 0.15.0 alpha/beta release? 

No.


I shall be able to try the manual commands 8-9 hours from now. Shall revert with the result thereafter.

**Comment by intelliant01** [comment]
Stopped Kolibri. Executed the above 2 commands.
Seems the migration did apply successfully -
```
mssadmin@server:~$ kolibri manage migrate morango 0018
WARNING  Option CHERRYPY_START in section [Server] is deprecated, please remove it from your options.ini file
INFO     Option DEBUG in section [Server] being overridden by environment variable KOLIBRI_DEBUG
INFO     Option DEBUG_LOG_DATABASE in section [Server] being overridden by environment variable KOLIBRI_DEBUG_LOG_DATABASE
INFO     Running Kolibri with the following settings: kolibri.deployment.default.settings.base
INFO     Importing 'tasks' module from django apps
INFO     Invoking command migrate morango 0018
Operations to perform:
  Target specific migration: 0018_auto_20210714_2216, from morango
Running migrations:
  Rendering model states... DONE
  Unapplying morango.0019_auto_20220113_1807... OK
mssadmin@server:~$ kolibri manage migrate
WARNING  Option CHERRYPY_START in section [Server] is deprecated, please remove it from your options.ini file
INFO     Option DEBUG in section [Server] being overridden by environment variable KOLIBRI_DEBUG
INFO     Option DEBUG_LOG_DATABASE in section [Server] being overridden by environment variable KOLIBRI_DEBUG_LOG_DATABASE
INFO     Running Kolibri with the following settings: kolibri.deployment.default.settings.base
INFO     Importing 'tasks' module from django apps
INFO     Invoking command migrate
Operations to perform:
  Apply all migrations: admin, analytics, auth, bookmarks, content, contenttypes, device, discovery, exams, kolibriauth, lessons, logger, morango, notifications, oidc_provider, sessions
Running migrations:
  Applying morango.0019_auto_20220113_1807... OK
```

Started Kolibri. Attempted a facility sync.
But the facility sync attempt still fails -

```
ERROR 2022-03-03 05:58:03,117 morango.sync.controller null value in column "transfer_stage_status" violates not-null constraint
DETAIL:  Failing row contains (831db069-cc51-43a1-b1c9-17fc1cafd690, f4dad04168d42689dd24713b3902595e, f, t, null, 46df154e-6ff4-40d1-b399-07bfd8dfcdad, 2022-03-03 00:28:03.048845+00, {"super": {}, "sub": {"f4dad04168d42689dd24713b3902595e": {"001a..., 0, {}, 2022-03-03 00:28:03.048771+00, 0, 0, initializing, null).

ERROR 2022-03-03 05:58:03,274 django.request Internal Server Error: /api/morango/v1/transfersessions/
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/kolibri/dist/morango/sync/operations.py", line 776, in handle
    transfer_session = transfer_sessions.get(**data)
  File "/usr/lib/python3/dist-packages/kolibri/dist/django/db/models/query.py", line 380, in get
    self.model._meta.object_name
morango.models.core.DoesNotExist: TransferSession matching query does not exist.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/kolibri/dist/django/db/backends/utils.py", line 64, in execute
    return self.cursor.execute(sql, params)
psycopg2.errors.NotNullViolation: null value in column "transfer_stage_status" violates not-null constraint
DETAIL:  Failing row contains (831db069-cc51-43a1-b1c9-17fc1cafd690, f4dad04168d42689dd24713b3902595e, f, t, null, 46df154e-6ff4-40d1-b399-07bfd8dfcdad, 2022-03-03 00:28:03.048845+00, {"super": {}, "sub": {"f4dad04168d42689dd24713b3902595e": {"001a..., 0, {}, 2022-03-03 00:28:03.048771+00, 0, 0, initializing, null).


The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/kolibri/dist/django/core/handlers/exception.py", line 41, in inner
    response = get_response(request)
  File "/usr/lib/python3/dist-packages/kolibri/dist/django/core/handlers/base.py", line 187, in _get_response
    response = self.process_exception_by_middleware(e, request)
  File "/usr/lib/python3/dist-packages/kolibri/dist/django/core/handlers/base.py", line 185, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/usr/lib/python3/dist-packages/kolibri/dist/django/views/decorators/csrf.py", line 58, in wrapped_view
    return view_func(*args, **kwargs)
  File "/usr/lib/python3/dist-packages/kolibri/dist/rest_framework/viewsets.py", line 116, in view
    return self.dispatch(request, *args, **kwargs)
  File "/usr/lib/python3/dist-packages/kolibri/dist/rest_framework/views.py", line 495, in dispatch
    response = self.handle_exception(exc)
  File "/usr/lib/python3/dist-packages/kolibri/dist/rest_framework/views.py", line 455, in handle_exception
    self.raise_uncaught_exception(exc)
  File "/usr/lib/python3/dist-packages/kolibri/dist/rest_framework/views.py", line 492, in dispatch
    response = handler(request, *args, **kwargs)
  File "/usr/lib/python3/dist-packages/kolibri/dist/morango/api/viewsets.py", line 392, in create
    raise context.error
  File "/usr/lib/python3/dist-packages/kolibri/dist/morango/sync/controller.py", line 236, in _invoke_middleware
    result = middleware(context)
  File "/usr/lib/python3/dist-packages/kolibri/dist/morango/registry.py", line 227, in __call__
    result = operation(context)
  File "/usr/lib/python3/dist-packages/kolibri/dist/morango/sync/operations.py", line 719, in __call__
    result = self.handle(context)
  File "/usr/lib/python3/dist-packages/kolibri/dist/morango/sync/operations.py", line 804, in handle
    transfer_session.save()
  File "/usr/lib/python3/dist-packages/kolibri/dist/django/db/models/base.py", line 808, in save
    force_update=force_update, update_fields=update_fields)
  File "/usr/lib/python3/dist-packages/kolibri/dist/django/db/models/base.py", line 838, in save_base
    updated = self._save_table(raw, cls, force_insert, force_update, using, update_fields)
  File "/usr/lib/python3/dist-packages/kolibri/dist/django/db/models/base.py", line 924, in _save_table
    result = self._do_insert(cls._base_manager, using, fields, update_pk, raw)
  File "/usr/lib/python3/dist-packages/kolibri/dist/django/db/models/base.py", line 963, in _do_insert
    using=using, raw=raw)
  File "/usr/lib/python3/dist-packages/kolibri/dist/django/db/models/manager.py", line 85, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
  File "/usr/lib/python3/dist-packages/kolibri/dist/django/db/models/query.py", line 1079, in _insert
    return query.get_compiler(using=using).execute_sql(return_id)
  File "/usr/lib/python3/dist-packages/kolibri/dist/django/db/models/sql/compiler.py", line 1112, in execute_sql
    cursor.execute(sql, params)
  File "/usr/lib/python3/dist-packages/kolibri/dist/django/db/backends/utils.py", line 64, in execute
    return self.cursor.execute(sql, params)
  File "/usr/lib/python3/dist-packages/kolibri/dist/django/db/utils.py", line 94, in __exit__
    six.reraise(dj_exc_type, dj_exc_value, traceback)
  File "/usr/lib/python3/dist-packages/kolibri/dist/django/utils/six.py", line 685, in reraise
    raise value.with_traceback(tb)
  File "/usr/lib/python3/dist-packages/kolibri/dist/django/db/backends/utils.py", line 64, in execute
    return self.cursor.execute(sql, params)
django.db.utils.IntegrityError: null value in column "transfer_stage_status" violates not-null constraint
DETAIL:  Failing row contains (831db069-cc51-43a1-b1c9-17fc1cafd690, f4dad04168d42689dd24713b3902595e, f, t, null, 46df154e-6ff4-40d1-b399-07bfd8dfcdad, 2022-03-03 00:28:03.048845+00, {"super": {}, "sub": {"f4dad04168d42689dd24713b3902595e": {"001a..., 0, {}, 2022-03-03 00:28:03.048771+00, 0, 0, initializing, null).
```

**Comment by bjester** [comment]
I'm not seeing that PostgreSQL 12 is available with Ubuntu 18.04. Could you share more about your configuration? You've configured Kolibri to access the database through the `options.ini`?

**Comment by bjester** [comment]
Even though the migration says it was applied, it didn't apply the migration's changes, clearly seen in the error raised during syncing. I've shared some SQL queries below that should correct this issue manually. I need additional information regarding your setup to ensure I attempt to reproduce this correctly

```sql
ALTER TABLE morango_transfersession ALTER COLUMN transfer_stage DROP NOT NULL;
ALTER TABLE morango_transfersession ALTER COLUMN transfer_stage_status DROP NOT NULL;
```

**Comment by intelliant01** [comment]
> I'm not seeing that PostgreSQL 12 is available with Ubuntu 18.04.

```
mssadmin@server:~$ locate bin/postgres
/usr/lib/postgresql/12/bin/postgres
mssadmin@server:~$ psql -V
psql (PostgreSQL) 12.3 (Ubuntu 12.3-1.pgdg18.04+1)
mssadmin@server:~$ /usr/lib/postgresql/12/bin/postgres -V
postgres (PostgreSQL) 12.3 (Ubuntu 12.3-1.pgdg18.04+1)
mssadmin@server:~$ dpkg --list |grep postgres
ii  postgresql                                      12+215.pgdg18.04+1
        all          object-relational SQL database (supported version)
ii  postgresql-12                                   12.3-1.pgdg18.04+1
        amd64        object-relational SQL database, version 12 server
ii  postgresql-client                               12+215.pgdg18.04+1
        all          front-end programs for PostgreSQL (supported version)
ii  postgresql-client-12                            12.3-1.pgdg18.04+1
        amd64        front-end programs for PostgreSQL 12
ii  postgresql-client-common                        190
        all          manager for multiple PostgreSQL client versions
ii  postgresql-common                               190
        all          PostgreSQL database-cluster manager
```



> Could you share more about your configuration? You've configured Kolibri to access the database through the `options.ini`?

```
mssadmin@server:~$ cat .kolibri/options.ini
[Cache]
 CACHE_BACKEND = redis
 CACHE_REDIS_MAXMEMORY = 404796621
 CACHE_REDIS_MAXMEMORY_POLICY = allkeys-lru
[Database]
 DATABASE_ENGINE = postgres
 DATABASE_NAME = kolibricloud14b15
 DATABASE_PASSWORD = *******
 DATABASE_USER = mssadmin
 DATABASE_HOST = 127.0.0.1
 DATABASE_PORT = 5432
[Paths]
 CONTENT_DIR = /home/mssadmin/.kolibri/content
 CONTENT_FALLBACK_DIRS = /home/mssadmin/.kolibri/,
[Urls]
[Deployment]
 URL_PATH_PREFIX = /
 HTTP_PORT = 8080
 ZIP_CONTENT_PORT = 8081
[Python]
[Server]
 CHERRYPY_START = False
[OIDCProvider]
 REQUIRE_CONSENT = False
[Tasks]
[Learn]
```

> I've shared some SQL queries below that should correct this issue manually.

Sync working after executing the queries. Thank you.

**Comment by intelliant01** [comment]
For postgresql-12, I may have used an installation procedure such as [this](https://computingforgeeks.com/install-postgresql-12-on-ubuntu/).

**Comment by bjester** [comment]
@intelliant01 Thank you for this additional context. Out of curiosity, is there a particular reason you used the PGDG repo to install postgres instead of version 10 available from the Ubuntu repos?

**Comment by bjester** [comment]
@intelliant01 Also, what method did you use to install `psycopg2`?

I've successfully reproduced your issue, but I'm not yet sure of the cause. I suspect an interaction with Django and Postgres 12 might be the cause

**Comment by intelliant01** [comment]
>  Out of curiosity, is there a particular reason you used the PGDG repo to install postgres instead of version 10 available from the Ubuntu repos?

I had just followed the sqlite to postgres migration instructions constructed by @lyw07 and shared by @jredrejo which had the following lines -

> install PostgreSQL (https://www.postgresql.org/download/)
> install psycopg2 pip install psycopg2
> 



> Also, what method did you use to install psycopg2

bash history shows this  -
```
  244  apt install python3-psycopg2
  694  venv/bin/pip3.6 install psycopg2
  695  cp -r venv//lib/python3.6/site-packages/psycopg2* /usr/lib/python3/dist-packages/kolibri/dist/
```

```
# pip show psycopg2
Name: psycopg2
Version: 2.8.5
Summary: psycopg2 - Python-PostgreSQL Database Adapter
Home-page: https://psycopg.org/
Author: Federico Di Gregorio
Author-email: fog@initd.org
License: LGPL with exceptions
Location: /usr/lib/python3/dist-packages
Requires:
```

```
# dpkg --list |grep psycopg2
ii  python3-psycopg2                                2.8.5-1~pgdg18.04+1                                                         amd64        Python 3 module for PostgreSQL
# dpkg-query -L  python3-psycopg2
/.
/usr
/usr/lib
/usr/lib/python3
/usr/lib/python3/dist-packages
/usr/lib/python3/dist-packages/psycopg2
/usr/lib/python3/dist-packages/psycopg2/__init__.py
/usr/lib/python3/dist-packages/psycopg2/_ipaddress.py
/usr/lib/python3/dist-packages/psycopg2/_json.py
/usr/lib/python3/dist-packages/psycopg2/_lru_cache.py
/usr/lib/python3/dist-packages/psycopg2/_psycopg.cpython-36m-x86_64-linux-gnu.so
/usr/lib/python3/dist-packages/psycopg2/_range.py
/usr/lib/python3/dist-packages/psycopg2/compat.py
/usr/lib/python3/dist-packages/psycopg2/errorcodes.py
/usr/lib/python3/dist-packages/psycopg2/errors.py
/usr/lib/python3/dist-packages/psycopg2/extensions.py
/usr/lib/python3/dist-packages/psycopg2/extras.py
/usr/lib/python3/dist-packages/psycopg2/pool.py
/usr/lib/python3/dist-packages/psycopg2/sql.py
/usr/lib/python3/dist-packages/psycopg2/tz.py
/usr/lib/python3/dist-packages/psycopg2-2.8.5.egg-info
/usr/lib/python3/dist-packages/psycopg2-2.8.5.egg-info/PKG-INFO
/usr/lib/python3/dist-packages/psycopg2-2.8.5.egg-info/dependency_links.txt
/usr/lib/python3/dist-packages/psycopg2-2.8.5.egg-info/top_level.txt
/usr/share
/usr/share/doc
/usr/share/doc/python3-psycopg2
/usr/share/doc/python3-psycopg2/changelog.Debian.gz
/usr/share/doc/python3-psycopg2/copyright
```

May be I/we tried both methods to eventually get the package inside `/usr/lib/python3/dist-packages/kolibri/dist/` .

```
# ls -l /usr/lib/python3/dist-packages/kolibri/dist/psycopg2
total 1292
-rw-r--r-- 1 root root     367 Aug  6  2020 compat.py
-rw-r--r-- 1 root root   14120 Aug  6  2020 errorcodes.py
-rw-r--r-- 1 root root    1420 Aug  6  2020 errors.py
-rw-r--r-- 1 root root    7122 Aug  6  2020 extensions.py
-rw-r--r-- 1 root root   44112 Aug  6  2020 extras.py
-rw-r--r-- 1 root root    4916 Aug  6  2020 __init__.py
-rw-r--r-- 1 root root    2967 Aug  6  2020 _ipaddress.py
-rw-r--r-- 1 root root    7296 Aug  6  2020 _json.py
-rw-r--r-- 1 root root    4261 Aug  6  2020 _lru_cache.py
-rw-r--r-- 1 root root    6319 Aug  6  2020 pool.py
-rwxr-xr-x 1 root root 1157552 Aug  6  2020 _psycopg.cpython-36m-x86_64-linux-gnu.so
drwxr-xr-x 2 root root    4096 Aug  6  2020 __pycache__
-rw-r--r-- 1 root root   17705 Aug  6  2020 _range.py
-rw-r--r-- 1 root root   14903 Aug  6  2020 sql.py
-rw-r--r-- 1 root root    4446 Aug  6  2020 tz.py

# ls -l /usr/lib/python3/dist-packages/psycopg2/
total 420
-rw-r--r-- 1 root root    367 Apr  6  2020 compat.py
-rw-r--r-- 1 root root  14120 Apr  6  2020 errorcodes.py
-rw-r--r-- 1 root root   1420 Apr  6  2020 errors.py
-rw-r--r-- 1 root root   7122 Apr  6  2020 extensions.py
-rw-r--r-- 1 root root  44112 Apr  6  2020 extras.py
-rw-r--r-- 1 root root   4916 Apr  6  2020 __init__.py
-rw-r--r-- 1 root root   2967 Apr  6  2020 _ipaddress.py
-rw-r--r-- 1 root root   7296 Apr  6  2020 _json.py
-rw-r--r-- 1 root root   4261 Apr  6  2020 _lru_cache.py
-rw-r--r-- 1 root root   6319 Apr  6  2020 pool.py
-rw-r--r-- 1 root root 264752 Apr  8  2020 _psycopg.cpython-36m-x86_64-linux-gnu.so
drwxr-xr-x 2 root root   4096 Jul  8  2020 __pycache__
-rw-r--r-- 1 root root  17705 Apr  6  2020 _range.py
-rw-r--r-- 1 root root  14903 Apr  6  2020 sql.py
-rw-r--r-- 1 root root   4446 Apr  6  2020 tz.py
```
This was almost 8 months ago, so can't say for sure why this was done this way.

I hope this is helpful.

**Comment by rtibbles** [comment]
Fixed in #9168 - this fix should be in the next 0.15 patch release.
